### PR TITLE
Incorporate invest-workbench into this repo

### LINKED ISSUE #877: Various models are not in sync with ui_config.js
This new parameter needs an entry in `renderer/ui_config.js`.

And maybe at the same time we can improve our tests to alert us of missing parameters like this. When we first created `ui_config.js`, I think the idea was that a model's UI should be fully functional even without an entry in the config. The intent of the config being to improve ordering & grouping of args.

But now it seems this is no longer true because leaving out a parameter from the config is our way of hiding a param from the UI (e.g. rec model's `hostname` & `port`). I think that's fine, it just means we should also improve our tests to alert us of this.

### PR DESCRIPTION
**Pull request by davemfish**
This PR adds all the workbench files into a `workbench` directory. It updates the invest GHA workflows to also test, build, and deploy the workbench. 

Changes on the workbench side (to remove now unnecessary scripts, etc) can also be reviewed in [this PR,](https://github.com/natcap/invest-workbench/pull/221) just to make it easier to review. So if you like, it's okay to only review files that are not in the `workbench` directory here. And those changes are limited to `Makefile` & two workflow `yml` files.

GHA:
===
Workbench builds happen within the existing invest build workflow and existing "Build InVEST App Binaries" jobs so that the same invest binaries can be used for the Qt app and the Workbench.

Workbench tests run in the existing invest test workflow, but in their own separate job because it's not necessary to run workbench tests on a matrix of python versions.

Workbench artifacts are deployed to the same GCS locations where invest artifacts are deployed, but to a `workbench` directory: 
`natcap-dev-build-artifacts/invest/davemfish/3.9.2.post1007+gbc9420eee/workbench`

Local Development
=============
Nothing much changes here. The `Makefile` is still basically just for invest and `package.json` is still there for the workbench. For example, 
`make test` still just runs invest tests. 

And familiar commands for the workbench can be issued from the `workbench` directory:
`cd workbench && yarn run test` 

Workbench development should be done from a shell with an active python env with `natcap.invest` installed, as the workbench expects `invest` to be on the PATH in devMode. 

Addresses #844 
Fixes #877 

## Checklist
- [ ] Updated HISTORY.rst (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the affected models' UIs (if relevant)

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ pytest
 pytest-subtests
 wheel>=0.27.0
 pypiwin32; sys_platform == 'win32'  # pip-only
-setuptools>=8.0
+setuptools>=8.0,<60.7.0 # https://github.com/pyinstaller/pyinstaller/issues/6564
```
Comment (emlys) [inline requirements-dev.txt:19]: Also from a different PR?

File: .github/workflows/binary-applications.yml [hunk 2]
```diff
@@ -206,25 +271,43 @@ jobs:
...
-              uses: actions/upload-artifact@v2
-              with:
-                  name: InVEST-mac-binaries
-                  path: dist/InVEST-*-mac.zip
+            # TODO: we're not building zips, right?
+            # - name: Upload binaries artifact
+            #   uses: actions/upload-artifact@v2
+            #   with:
+            #       name: InVEST-mac-binaries
+            #       path: dist/InVEST-*-mac.zip
```
Comment (emlys) [inline .github/workflows/binary-applications.yml:289]: Yeah, this step can be deleted.

File: Makefile [hunk 3]
```diff
@@ -34,7 +34,7 @@ ifeq ($(OS),Windows_NT)
 	# widely available on Windows now, especially through git-bash
 	SHELL := /usr/bin/bash
 	CONDA := conda.bat
-	BASHLIKE_SHELL_COMMAND := $(SHELL) -c
+	BASHLIKE_SHELL_COMMAND := '$(SHELL)' -c
```
Comment (dcdenu4) [inline Makefile:37]: Nice, I also noticed this in something I'm working on

File: Makefile [hunk 4]
```diff
@@ -382,20 +385,21 @@ codesign_mac:
...
-	"$(SIGNTOOL)" sign -fd SHA256 -f $(BUILD_DIR)/$(CERT_FILE) -p $(CERT_PASS) $(BIN_TO_SIGN)
-	"$(SIGNTOOL)" timestamp -tr http://timestamp.sectigo.com -td SHA256 $(BIN_TO_SIGN)
+	"$(SIGNTOOL)" sign -fd SHA256 -f $(BUILD_DIR)/$(CERT_FILE) -p $(CERT_PASS) $(BIN_TO_SIGN) $(WORKBENCH_BIN_TO_SIGN)
+	"$(SIGNTOOL)" timestamp -tr http://timestamp.sectigo.com -td SHA256 $(BIN_TO_SIGN) $(WORKBENCH_BIN_TO_SIGN)
 	$(RM) $(BUILD_DIR)/$(CERT_FILE)
 	@echo "Installer was signed with signtool"
 
 deploy:
-	-(cd $(INVEST_BINARIES_DIR) && $(ZIP) -r ../$(INVEST_BINARIES_DIR_ZIP) .)
+# 	-(cd $(INVEST_BINARIES_DIR) && $(ZIP) -r ../$(INVEST_BINARIES_DIR_ZIP) .)
```
Comment (dcdenu4) [inline Makefile:398]: Can this and the related commented out line above be removed?

**Review by dcdenu4 — CHANGES_REQUESTED** [review]
Hey @davemfish , this is looking good! Excited to pull these changes locally soon. Everything from this PR and the related draft PR over on the workbench repo make sense to me. As Emily noted there are a few things in here that were seemingly taken care of in past PRs.

**Comment by davemfish** [comment]
@emlys & @dcdenu4 I pushed updates to remove those bits of commented code, and to merge in the latest `main`. So I think those things address all your comments.

I see that there is still a README in `workbench/internationalization`, but it does seem to have some unique content relative to the invest readme so I think I'll leave it as-is and we can remove redundancy later if there is some?

There was a leftover workbench script for code signing that I just removed too.

**Review by phargogh — APPROVED** [review]
Very exciting to see the two repos coming together!  I don't see anything that hasn't already been commented on.

**Review by emlys — APPROVED** [review]

**Pull request merged**